src/spdx_tools/spdx/model/document.py

```diff
@@ -81,3 +81,18 @@ def __init__(
         relationships = [] if relationships is None else relationships
         extracted_licensing_info = [] if extracted_licensing_info is None else extracted_licensing_info
         check_types_and_set_values(self, locals())
+
+
+def document_cache(func):
+    cache = {}
+
+    def cached_function(document: Document):
+        key = id(document)
+        if key in cache.keys():
```

Comment:
Or in other words: does the cache key `id(document)` change, if I mutate some stuff inside the document?

Maybe a less global caching would be more appropriate. So that its scope is controlled in a way so that no mutation can happen in between.

Reply:
I don't think so. From the [docs](https://docs.python.org/3/library/functions.html#id):

```
id(object)

    Return the “identity” of an object. This is an integer which is guaranteed to be unique and constant for this object during its lifetime.
Two objects with non-overlapping lifetimes may have the same [id()](https://docs.python.org/3/library/functions.html#id) value.
``

Reply:
I think that this is exactly the problem I am afraid of

```python
>>> a = {1}
>>> id(a)
139842939239360
>>> a.add(2)
>>> a
{1, 2}
>>> id(a)
139842939239360
```

The object has the same ID and it does not change even if its content is mutated. So it is maybe not a good cache key.